### ISSUE TITLE
more on method parameters

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#2085]
should give syntax error,

```abap
CLASS bar DEFINITION.
  PUBLIC SECTION.
    METHODS run.
    METHODS call IMPORTING bar TYPE i.
ENDCLASS.

CLASS bar IMPLEMENTATION.
  METHOD call.
    RETURN.
  ENDMETHOD.

  METHOD run.
    call( ).
  ENDMETHOD.
ENDCLASS.
```

**Comment by larshp** [comment]
CHANGING, should give syntax error,

```abap
CLASS bar DEFINITION.
  PUBLIC SECTION.
    METHODS run.
    METHODS call CHANGING bar TYPE i.
ENDCLASS.

CLASS bar IMPLEMENTATION.
  METHOD call.
    RETURN.
  ENDMETHOD.

  METHOD run.
    call( ).
  ENDMETHOD.
ENDCLASS.
```

**Comment by FreHu** [comment]
writing to importing parameters should be an error (unless they are by value)

**Comment by larshp** [comment]
@FreHu yea, open as a separate issue?
